corporate/lib/stripe_event_handler.py

```diff
@@ -61,6 +63,20 @@ def wrapper(
     return wrapper
 
 
+def get_billing_session(
+    customer: Customer, user_id: Optional[str]
+) -> Union[RealmBillingSession, RemoteRealmBillingSession, RemoteServerBillingSession]:
+    if customer.remote_realm is not None:  # nocoverage
+        return RemoteRealmBillingSession(customer.remote_realm)
+    elif customer.remote_server is not None:  # nocoverage
+        return RemoteServerBillingSession(customer.remote_server)
+    else:
+        assert user_id is not None
+        assert customer.realm is not None
+        user = get_active_user_profile_by_id_in_realm(int(user_id), customer.realm)
+        return RealmBillingSession(user)
```

Comment:
I think we might want to name this `get_billing_session_for_stripe_webhook`; it's a bit of a system job component.

A related note is that I wonder if we should replace calling `update_or_create_stripe_customer` with a payment method (the main thing using this billing session) with just a function that obviously only can update the payment method -- I'm pretty sure we'll never be creating a customer in this code path (they should exist already), and the APIs making that clearer would help.

Reply:
Makes sense but I had a lot of test failure trying to make that change so we can focus on that later. Opened #28002 to track it.